libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -162,6 +162,56 @@ bool DocStringTagParser::visit(ErrorDefinition const& _error)
 	return true;
 }
 
+bool DocStringTagParser::visit(InlineAssembly const& _assembly)
+{
+	if (!_assembly.documentation())
+		return true;
+	StructuredDocumentation documentation{-1, _assembly.location(), _assembly.documentation()};
+	ErrorList errors;
+	ErrorReporter errorReporter{errors};
+	auto docTags = DocStringParser{documentation, errorReporter}.parse();
+
+	if (!errors.empty())
+	{
+		SecondarySourceLocation ssl;
+		for (auto const& error: errors)
+			if (error->comment())
+				ssl.append(
+					*error->comment(),
+					error->sourceLocation() ? *error->sourceLocation() : _assembly.location()
+				);
+		m_errorReporter.warning(
+			7828_error,
+			_assembly.location(),
+			"Inline assembly has invalid natspec documentation.",
+			ssl
+		);
+	}
+
+	for (auto const& [tagName, tagValue]: docTags)
+	{
+		if (tagName == "solidity")
+		{
+			if (tagValue.content == "memory-safe-assembly")
```

Comment:
Do we already want to split this into words?

Reply:
Not sure what you mean exactly... make it ``memory safe assembly`` or ``assembly: memory-safe`` or something? Or just split at the hyphens here in case we want to add more of them?

Reply:
Split the tag value at whitespace so that we get an array. Loop over the array. Create a warning for any unknown value and handle the value `memory-safe-assembly` as here.